In [2]:
import pandas as pd

data = {
    "A": [12, 15, 14, 16, 18, 17, 19, 20, 21, 22, 23, 24, 25, 26, 27],
    "B": [35, 34, 36, 38, 40, 39, 41, 42, 43, 44, 45, 46, 47, 48, 49],
    "C": [102, 98, 100, 105, 110, 108, 112, 115, 118, 120, 122, 125, 128, 130, 132],
    "D": [5.2, 5.5, 5.4, 5.6, 5.8, 5.7, 5.9, 6.0, 6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7],
    "E": [200, 210, 205, 215, 220, 218, 225, 230, 235, 240, 245, 250, 255, 260, 265]
}

df = pd.DataFrame(data)
df

A   B    C    D    E
0   12  35  102  5.2  200
1   15  34   98  5.5  210
2   14  36  100  5.4  205
3   16  38  105  5.6  215
4   18  40  110  5.8  220
5   17  39  108  5.7  218
6   19  41  112  5.9  225
7   20  42  115  6.0  230
8   21  43  118  6.1  235
9   22  44  120  6.2  240
10  23  45  122  6.3  245
11  24  46  125  6.4  250
12  25  47  128  6.5  255
13  26  48  130  6.6  260
14  27  49  132  6.7  265

##Range

In [7]:
A_range = df["A"].max() - df["A"].min()


print(f"Range of A: {A_range}")


Range of A: 15


##Variance:

In [4]:
A = df["A"].to_list()

mean_A = sum(A) / len(A)
print(f"Mean of A: {mean_A}")

sum_A = 0
for x in A:
    sum_A += (x - mean_A) ** 2

print(f"Sum of squared deviations (A): {sum_A}")

variance_A = sum_A / (len(A) - 1)
print(f"Sample Variance of A: {variance_A}")

Mean of A: 19.933333333333334
Sum of squared deviations (A): 294.93333333333334
Sample Variance of A: 21.066666666666666


##Standard Devation

In [5]:
sd_A = variance_A ** 0.5
print(f"Standard Deviation (A): {sd_A}")

Standard Deviation (A): 4.589843860815601


##Z Score

In [6]:
value = 16
z_A = (value - mean_A) / sd_A
print(f"Z-score (A): {z_A}")

Z-score (A): -0.8569645183168371


##Conclusion:

Range is 15, mean is 19.93, variance is 21.06 and standard deviation is 4.58.
The Z-score value is −0.85, which shows that the selected value (16) is slightly lower than the average. This value is normal and not very far from the mean